endpoints/openrtb2/auction_test.go

```diff
@@ -5618,6 +5618,92 @@ func TestValidateOrFillCookieDeprecation(t *testing.T) {
 	}
 }
 
+func TestSetGPCImplicitly(t *testing.T) {
+	testCases := []struct {
+		description  string
+		header       string
+		expectedGPC  string
+		expectError  bool
+		regs         *openrtb2.Regs
+		expectedRegs *openrtb2.Regs
+	}{
+		{
+			description: "regs_ext_gpc_not_set_and_header_is_1",
+			header:      "1",
+			expectedGPC: "1",
+			expectError: false,
+			regs: &openrtb2.Regs{
+				Ext: []byte(`{}`),
+			},
+			expectedRegs: &openrtb2.Regs{
+				Ext: []byte(`{"gpc":"1"}`),
+			},
+		},
+		{
+			description: "sec_gpc_header_not_set_gpc_should_not_be_modified",
```

Comment:
I suggest covering the following test cases:
`regs_ext_gpc_not_set_and_header_not_set`
`regs_ext_gpc_not_set_and_header_is_1`
`regs_ext_gpc_not_set_and_header_not_1`
`regs_ext_gpc_is_1_and_header_is_1`
`regs_ext_gpc_is_1_and_header_not_1`
`regs_ext_other_data_and_header_is_1`
`regs_ext_nil_and_header_is_1`
`regs_nil_and_header_is_1`